File: DSA/Manav-Rajput/Day68_question2.java
 class Solution {
    public List<Integer> preorderTraversal(TreeNode root) {
        List<Integer> result = new ArrayList<>();
        preorderHelper(root, result);
        return result;
    }
    
    private void preorderHelper(TreeNode node, List<Integer> result) {
        if (node == null) {
            return;
        }
        
        // Visit the root node
        result.add(node.val);
        
        // Recurse on the left subtree
        preorderHelper(node.left, result);
        
        // Recurse on the right subtree
        preorderHelper(node.right, result);
    }
}

File: DSA/Arshdeep-Kaur/Day 68/Day68_Question2.cpp
class Solution {
public:
    void preOrder(TreeNode* node, vector<int> &ans) {
        //base case 
        if (node == NULL) {
            return;
        }

        ans.push_back(node -> val);
        preOrder(node -> left, ans);
        preOrder(node -> right, ans);
    }

    vector<int> preorderTraversal(TreeNode* root) {
        vector<int> ans;

        preOrder(root, ans);
        return ans;
    }
};
